### ISSUE TITLE
Vulnerabilities of NPM package `serve` incorrectly reported against `@webpack-cli/serve`

### ISSUE THREAD

**Issue by nil4** [DependencyTrack/dependency-track#804]
### Current Behavior:

Using DependencyTrack 3.8.0, multiple vulnerabilities relevant for the NPM `serve` package are incorrectly attributed to a different package, `@webpack-cli/serve`. 

### Steps to Reproduce:

In an empty folder, create a `package.json` file with the following content:

```json
{
  "name": "repro",
  "version": "1.0.0",
  "private": true,
  "dependencies": {
    "webpack-cli": "^4.0.0"
  }
}
```

Run `npm install` to install the `webpack-cli` NPM package and its dependencies.

Run `npm install --global @cyclonedx/bom` to install the CycloneDX BOM tool for NodeJS.

Run `cyclonedx-bom -ns` in the folder containing the `package.json` file. A `bom.xml` file is produced, listing dependencies of `webpack-cli`. 

Review the `bom.xml` file contents and verify that:
- references to `pkg:npm/%40webpack-cli/serve@1.0.1` are present, but
- *no* references to `pkg:npm/serve@` are present

Upload the `bom.xml` file to DependencyTrack and wait for analysis results.

### Expected Behavior:

Only vulnerabilities relevant to the packages included in the uploaded `bom.xml` are reported.

### Observed Behavior:

9 vulnerabilities, relevant to the [`serve`](https://www.npmjs.com/package/serve) NPM package, which is *not* referenced by this project, are reported but incorrectly attributed to [`@webpack-cli/serve`](https://www.npmjs.com/package/@webpack-cli/serve):

![image](https://user-images.githubusercontent.com/139133/96363990-3cc02c00-1138-11eb-8377-8b67affecdb2.png)

These vulnerabilities have links to various HackerOne reports, which clearly indicate that the `serve` NPM package is vulnerable, e.g.:
- https://hackerone.com/reports/308721
- https://hackerone.com/reports/330650
- https://hackerone.com/reports/398285
- https://hackerone.com/reports/358645
- https://hackerone.com/reports/358641
- https://hackerone.com/reports/308721
- https://hackerone.com/reports/330724

### Environment:
  
  - Dependency-Track Version: 3.8.0
  - Distribution: Executable WAR
  - BOM Format & Version: XML, 1.2
  - Database Server: MSSQL
  - Browser: all (N/A)

### Additional Details:

The `package.json`, `package-lock.json` and `bom.xml` files, obtained locally for this repro, are attached for reference: [dt-serve-repro.zip](https://github.com/DependencyTrack/dependency-track/files/5397313/dt-serve-repro.zip)

  - NodeJS (`node -v`): 14.14.0 (x64) Windows
  - NPM (`npm -v`): 7.0.2
  - CycloneDX/BOM version: 2.0.1

**Comment by nil4** [comment]
Here is a much smaller repro with just two dependencies. 
This `package.json` file:

```json
{
  "name": "repro2",
  "version": "2.0.0",
  "private": true,
  "dependencies": {
    "@types/jquery": "^2.0.34",
    "@types/lodash": "^4.14.106"
  }
}
```

Produces the `bom.xml` below. Note that neither jquery and lodash packages are *actually* part of the BOM:

```xml
<?xml version="1.0" encoding="utf-8"?>
<bom xmlns="http://cyclonedx.org/schema/bom/1.2" version="1">
    <metadata>
        <timestamp>2020-10-18T11:28:35.000Z</timestamp>
        <tools>
            <tool>
                <vendor>CycloneDX</vendor>
                <name>Node.js module</name>
                <version>2.0.1</version>
            </tool>
        </tools>
        <component type="library" bom-ref="pkg:npm/repro2@2.0.0">
            <name>repro2</name>
            <version>2.0.0</version>
            <purl>pkg:npm/repro2@2.0.0</purl>
        </component>
    </metadata>
    <components>
        <component type="library" bom-ref="pkg:npm/%40types/jquery@2.0.56">
            <group>@types</group>
            <name>jquery</name>
            <version>2.0.56</version>
            <description>
                <![CDATA[TypeScript definitions for jQuery]]>
            </description>
            <licenses>
                <license>
                    <id>MIT</id>
                </license>
            </licenses>
            <purl>pkg:npm/%40types/jquery@2.0.56</purl>
            <externalReferences>
                <reference type="website">
                    <url>https://github.com/DefinitelyTyped/DefinitelyTyped#readme</url>
                </reference>
                <reference type="issue-tracker">
                    <url>https://github.com/DefinitelyTyped/DefinitelyTyped/issues</url>
                </reference>
                <reference type="vcs">
                    <url>git+https://github.com/DefinitelyTyped/DefinitelyTyped.git</url>
                </reference>
            </externalReferences>
        </component>
        <component type="library" bom-ref="pkg:npm/%40types/lodash@4.14.162">
            <group>@types</group>
            <name>lodash</name>
            <version>4.14.162</version>
            <description>
                <![CDATA[TypeScript definitions for Lo-Dash]]>
            </description>
            <licenses>
                <license>
                    <id>MIT</id>
                </license>
            </licenses>
            <purl>pkg:npm/%40types/lodash@4.14.162</purl>
            <externalReferences>
                <reference type="website">
                    <url>https://github.com/DefinitelyTyped/DefinitelyTyped#readme</url>
                </reference>
                <reference type="issue-tracker">
                    <url>https://github.com/DefinitelyTyped/DefinitelyTyped/issues</url>
                </reference>
                <reference type="vcs">
                    <url>git+https://github.com/DefinitelyTyped/DefinitelyTyped.git</url>
                </reference>
            </externalReferences>
        </component>
    </components>
</bom>
```

Uploading the BOM to DT reports jQuery and lodash vulnerabilities:

![image](https://user-images.githubusercontent.com/139133/96366294-1a81da80-1147-11eb-8c20-f1728d6303bb.png)

It's possible that I may have missed something, but the behavior does not look expected.

**Comment by stevespringett** [comment]
This ticket likely describes behavior (scoped vs non-scoped package with the same name), as detailed here: https://github.com/DependencyTrack/dependency-track/issues/672

**Comment by nil4** [comment]
Confirmed fixed in 4.0.0-beta.3 -- thank you @stevespringett
